### PR TITLE
Fix vectorization in Poincare Ball

### PR DESCRIPTION
**Pull request by tgeral68**

### CODE REVIEW

**Comment by tgeral68** [comment]
There are some missing functionalities with the Pytorch backend for broadcasting in einsum, particularly `gs.einsum('j...,j...->j...',a, b)`  do not works having `a.shape=(1,d)` and `b.shape=n,1` in pytorch only (and here, it is a possible case).  

But I still can change the function to use einsum but keeping some repeat.

**Comment by ninamiolane** [comment]
How about: `gs.einsum('...j,...k->...j', a, b)`? I've started working on the `einsum` function in the backends, to extend it beyond its capacities for pytorch and tf.

File: geomstats/geometry/poincare_ball.py [hunk 1]
```diff
@@ -290,7 +298,7 @@ def retraction(self, tangent_vec, base_point):
         return base_point - retraction_factor * tangent_vec
 
     def inner_product_matrix(self, base_point=None):
-        """Compute the inner product matrix, independent of the base point.
+        """Compute the inner product matrixx.
```
Comment (ninamiolane) [inline geomstats/geometry/poincare_ball.py:301]: NIT: Typo.

File: tests/test_poincare_ball.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def setUp(self):
         self.hyperboloid_manifold = Hyperboloid(2)
         self.hyperboloid_metric = self.hyperboloid_manifold.metric
 
-    @geomstats.tests.np_and_pytorch_only
+    @geomstats.tests.np_only
```
Comment (ninamiolane) [inline tests/test_poincare_ball.py:26]: What happened for the pytorch test here?

Comment (tgeral68) [reply tests/test_poincare_ball.py:26]: My bad, it should works with pytorch.

**Pull request merged**